python/cudf/cudf/core/dataframe.py

```diff
@@ -7945,7 +7945,7 @@ def _setitem_with_dataframe(
             "Number of Input Columns must be same replacement Dataframe"
         )
 
-    if not input_df.index.equals(replace_df.index):
+    if not input_df.index.equals(replace_df.index) and len(input_df) != 0:
```

Comment:
Can we add a test to cover this?

Reply:
Yes I'm working on it now, will push as soon as it's ready.